scripts/validate-packages.py

```diff
@@ -133,6 +138,22 @@ def _validate_revision(given_package, revision, path):
                  'only supported when minDcosReleaseVersion is greater than '
                  '1.8.')
 
+def _validate_resource_assets(resource_json):
+    all_good = True
+    for (type, uri) in resource_json.get('assets',{}).get('uris',{}).items():
+
+        # Check for non-https assets
+        if (urlparse(uri).scheme != 'https'):
+            all_good = False
+            eprint("\n\t\t\tWARNING non-https link: {}".format(uri), end='')
+
+        # check for links on downloads downloads.mesosphere.com that do not point
+        # to a dedicated sub dir.
+        if (urlparse(uri).hostname == 'downloads.mesosphere.com' and not urlparse(uri).path.startswith('/universe/assets/') ):
+            all_good = False
+            eprint("\n\t\t\tWARNING downloads.mesosphere.com asset in stray directory: {}".format(uri), end='')
+
+    return all_good
```

Comment:
Regarding the two warnings above, we should turn them to errors and `sys.exit`. Developers are not going to notice those warning and it wont have the effect you want.